src/modules/commander/Commander.cpp

```diff
@@ -4028,6 +4028,12 @@ void Commander::estimator_check(bool *status_changed)
 	if (_estimator_status_sub.update()) {
 		const estimator_status_s& estimator_status = _estimator_status_sub.get();
 
+		// Check for a magnetomer fault and notify the user
+		if (!_mag_sensor_failed && (estimator_status.control_mode_flags & (1 << estimator_status_s::CS_MAG_FAULT))) {
+			_mag_sensor_failed = true;
```

Comment:
Are you going to use this flag for anything else in commander? If not, then you can drop it and simply store the previous mag fault state right before the estimator status update (linue 4028).

Reply:
OK, will do that.

Reply:
I realised that the problem with that approach is that I only want to send a message telling someone to check a sensor calibration once per flight, not every time the flag goes true.

Reply:
How annoying should this be? We wouldn't want to send it periodically (at a low rate) or trigger a failsafe?

Reply:
For Fixed Wing there is no reason to failsafe since the vehicle can fly and land completely safely without a magnetometer. 

I think announcing the warning only once when the EKF stops using the magnetometer is fine.  We've shown that fixed wing vehicles with badly calibrated compass can even auto-takeoff completely fine (a little extra wandering during initial climb, but not too bad).  A user would get another warning on the next flight if it was still out of calibration.

If multiple or periodic warnings are considered important, then some thought should go into the appropriate level of nagging.  once per minute?  once per 5 minutes?

Reply:
> trigger a failsafe?

 I think that is an important feature for MC. Unlike FW, an MC can't safely do position control without a mag. I've stumbled across this before that it is possible to fly in position control without a valid mag (https://github.com/PX4/ecl/issues/479, https://github.com/PX4/Firmware/issues/9901) but this is probably outside the scope of this issue.

Reply:
For reference the log in https://github.com/PX4/ecl/issues/479 didn't raise the same mag fault.